perception_msgs_rendering/src/rendering/object_state/object_state.cpp

```diff
@@ -240,6 +271,17 @@ void ObjectState::setObjectStateVizDefault(const perception_msgs::msg::ObjectSta
   }
 
   if (visualize_mesh_) {
+    // Clean up any previously created mesh node and its attached objects to avoid leaks
+    if (mesh_node_) {
+      // Detach and destroy all attached movable objects (e.g., entities)
+      while (mesh_node_->numAttachedObjects() > 0) {
+        Ogre::MovableObject* mo = mesh_node_->getAttachedObject(0);
+        mesh_node_->detachObject(mo);
+        scene_manager_->destroyMovableObject(mo);
+      }
+      scene_manager_->destroySceneNode(mesh_node_);
+      mesh_node_ = nullptr;
+    }
     //load mesh to render based on classification
```

Comment:
```suggestion
    // load mesh to render based on classification
```

In general, harmonize all comments!